diff -up libmikmod-3.2.0/config.h.in~ libmikmod-3.2.0/config.h.in
diff -up libmikmod-3.2.0/configure.in~ libmikmod-3.2.0/configure.in
--- libmikmod-3.2.0/configure.in~	2012-06-01 15:15:29.000000000 +0200
+++ libmikmod-3.2.0/configure.in	2013-04-06 12:43:30.200076725 +0200
@@ -634,66 +634,6 @@ then
 	fi
 fi
 
-# If compiling with gcc, use adequate optimization flags
-if test $ac_cv_prog_gcc = yes
-then
-	# On at least x86 platforms, gcc 2.7.2.1 and earlier won't work if -O3 (or
-	# -finline-functions) are used. Versions 2.7.2.3, 2.8.x and egcs are ok
-	# (didn't test 2.7.2.2).
-	# Until there's an easy way to catch broken inlining, we choose flags known
-	# to work correctly depending of the compiler version.
-	AC_CACHE_CHECK([if inlining functions is safe],
-	libmikmod_cv_gcc_inline_safe,
-	AC_TRY_RUN([
-int main()
-{
-#if defined __GNUC__ && (__GNUC__ > 2 || __GNUC__ == 2 && __GNUC_MINOR__ > 7)
-	exit(0);
-#else
-	exit(1);
-#endif
-}],
-	libmikmod_cv_gcc_inline_safe=yes,
-	libmikmod_cv_gcc_inline_safe=no,
-	libmikmod_cv_gcc_inline_safe=no))
-	if test $libmikmod_cv_gcc_inline_safe = yes
-	then
-		CFLAGS="$CFLAGS -finline-functions"
-	else
-		CFLAGS="$CFLAGS -fno-strength-reduce"
-	fi
-	CFLAGS="$CFLAGS -funroll-loops -ffast-math"
-	if test $libmikmod_debug = yes
-	then
-		CFLAGS="$CFLAGS -Wall -Werror"
-	else
-		CFLAGS="$CFLAGS -Wall"
-	fi
-
-	# pgcc 2.95.2 appears not to be able to compile libmikmod, although regular
-	# gcc works fine. Issue a warning if the compiler is pgcc, until a reliable
-	# way to detect flawed version (or, better, a reliable workaround) is
-	# found.
-dnl do not cache this test - better rechecking every time
-	AC_MSG_CHECKING(if compiler is pgcc)
-	if ($CC -v 2>&1 | grep ^pgcc > /dev/null) 2>/dev/null
-	then
-		libmikmod_gcc_is_pgcc=yes
-	else
-		libmikmod_gcc_is_pgcc=no
-	fi
-	AC_MSG_RESULT($libmikmod_gcc_is_pgcc)
-	if test $libmikmod_gcc_is_pgcc = yes
-	then
-		echo "
-*** Version 2.95.2 of this compiler, and perhaps others, are unable to
-*** compile libmikmod. If compilation fails for playercode/virtch.c
-*** around line 650, with the error ``internal error--insn does not
-*** satisfy its constraints'', then you'll have to use another compiler.
-" >&2
-	fi
-fi
-
 if test x$libmikmod_threads != xno
 then
 	AC_DEFINE(HAVE_PTHREAD)
